### ISSUE TITLE
.swiftlint.yml Is Not Recognized When SwiftLint Is Added As Binary

### ISSUE THREAD

**Issue by tamarnachmany** [realm/SwiftLint#257]
I'm attempting to integrate SwiftLint by adding the library as a binary in an app and the configuration of my .swiftlint.yml file is not being respected when I run linting. 

Is there a reason why this approach would not work?

I validated the yaml on http://www.yamllint.com -- it's correct -- and the yaml file is in the same directory where I am doing the linting.

**Comment by jpsim** [comment]
What does `swiftlint version` print? How about passing the YAML path directly via `swiftlint lint --config ...`? Does SwiftLint log `Loading configuration from ...` when you run it? If you post the contents of your config file here I may be able to debug further.

**Comment by tamarnachmany** [comment]
* SwiftLint version prints ```0.4.0.```
* ```Loading configuration from...``` is not logged when It's run.
* Running ```swiftlint lint --config "${SRCROOT}/.swiftlint.yml``` does not configure correctly either.

**Comment by jpsim** [comment]
Thanks for the additional info! I'd recommend you build & run SwiftLint from source in Xcode, which will give you access to LLDB to better debug why this isn't working:

```shell
git clone https://github.com/realm/SwiftLint.git
cd SwiftLint
git submodule update --init --recursive
open SwiftLint.xcworkspace
```

Then edit the `swiftlint` build scheme to use a custom working directory pointing to where your project is. Build & run, which should do the same as running `swiftlint` from that directory in a Terminal session.

You could set a breakpoint at https://github.com/realm/SwiftLint/blob/0.4.0/Source/SwiftLintFramework/Models/Configuration.swift#L101, and then print out `fullPath` and `NSFileManager.defaultManager().currentDirectoryPath` to see if they're what you expect.

**Comment by jpsim** [comment]
If you share a project that reproduces the issue you're seeing (or if your project is open source), let me know where I can find it so I can debug this with you too.

**Comment by tamarnachmany** [comment]
Thanks JP!

This has worked for me when running SwiftLint from source so I expect that if I try what you suggested above it will work correctly. I will play around with it.

I'm going to quickly throw up an implementation of how I have this set up in a sample project and I'll post a link here.

**Comment by jpsim** [comment]
Maybe your Xcode build phase is the issue? `swiftlint lint --config "${SRCROOT}/.swiftlint.yml`

It looks like you're missing a closing quote at the end there.

**Comment by tamarnachmany** [comment]
Oop. There's actually a quote in there I just didn't paste it in here by accident.

**Comment by tamarnachmany** [comment]
https://github.com/tamarnachmany/swiftlint-config-debugging

This should recreate my current configuration -- only difference is that this sample project does not include or exclude directories, whereas the app excludes our 'Pods' directory from linting.

**Comment by jpsim** [comment]
Thanks for posting that. The problem is that some rules are disabled twice, which is a format issue. Running `swiftlint` from the root of your repo yields the following output: `config error: 'function_body_length' is listed 2 times`. So just remove one of those from `excluded_rules` and you'll be ok!

**Comment by tamarnachmany** [comment]
Thanks a lot! 
I hadn't seen that config error error - I'll definitely use that as a debugging tool next time.

Also, I know someone is working on adding something to the project which validates the .swiftlint.yml file. We could add something to check for repeat disabled rule declarations in that PR (or a related one) potentially.

**Comment by jpsim** [comment]
> We could add something to check for repeat disabled rule declarations in that PR (or a related one) potentially.

This *is* that check. See https://github.com/realm/SwiftLint/blob/0.4.0/Source/SwiftLintFramework/Models/Configuration.swift#L66-L80

**Comment by tamarnachmany** [comment]
:ok_hand:
